### ISSUE TITLE
Convert namespace from aspect to argument in service and store layer

### ISSUE THREAD

**Issue by pmuir** [fabric8-ui/fabric8-runtime-console#143]
Right now the namespace is injected and stored in both the stores and the services, which makes them very hard to reuse as composable services.

This happens via the NamespacedResourceService and NamespacedResourceStore.

Instead of injecting the namespace into a parent class, it would be better if operations like loadAll took an argument of namespace: Observable<string> allowing the consumer to explicitly choose the namespace.

**Comment by jstrachan** [comment]
relates to https://github.com/fabric8-ui/fabric8-runtime-console/issues/146#issuecomment-288066353 - I wonder if we can use DI to inject the namespace from the route; using route specific injections? This would avoid the need to use `Observable<Params>` to watch for namespace changes?

**Comment by jstrachan** [comment]
also if we injected the namespace using the router; we could then look at adding more caching between related pages in the same namespace maybe?

**Comment by pmuir** [comment]
I just want to be able to call the service layer, passing in the namespace to look at.

**Comment by jstrachan** [comment]
yeah - we can maybe just put the ActiveRoute thingy on the _observer producer_ objects and keep the services super simple?

**Comment by pmuir** [comment]
I think if the observer producers *only* extracted the namespace from the activeroute and then called in to the service layer, that would work well. 

There is currently some useful functionality in the observer producer (aka store) layer (e.g. attaching a watcher, the loading stuff) that should move up in to the service layer - IOW the observer producer layer is simply a really thin facade that sets the right namespace.

WDYT?

**Comment by jstrachan** [comment]
yeah - sounds fine to me. Plus folks could compose the observable producers maybe with a different `Observable<string>` for the current namespace to the `ActiveRoute params` thingy (e.g. if it wanted to use a different namespace to the one from the route)
